atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/AtlasBackupService.java

```diff
@@ -202,24 +218,47 @@ public Set<AtlasService> completeBackup(Set<AtlasService> atlasServices) {
         Map<AtlasService, CompletedBackup> successfulBackups = KeyedStream.of(response.getSuccessfulBackups())
                 .mapKeys(token -> namespaceToServices.get(token.getNamespace()))
                 .collectToMap();
-        KeyedStream.stream(successfulBackups).forEach(this::storeCompletedBackup);
+        Set<AtlasService> successfullyStoredBackups = storeCompletedBackups(successfulBackups);
 
-        if (response.getSuccessfulBackups().size() < atlasServicesWithInProgressBackups.size()) {
-            Set<AtlasService> successfulAtlasServices = successfulBackups.keySet();
+        if (successfullyStoredBackups.size() < atlasServicesWithInProgressBackups.size()) {
             Set<AtlasService> failedAtlasServices =
-                    Sets.difference(atlasServicesWithInProgressBackups, successfulAtlasServices);
+                    Sets.difference(atlasServicesWithInProgressBackups, successfullyStoredBackups);
             log.error(
                     "Backup did not complete successfully for all atlasServices. Check TimeLock logs to debug.",
                     SafeArg.of("failedAtlasServices", failedAtlasServices),
-                    SafeArg.of("successfulAtlasServices", successfulAtlasServices),
+                    SafeArg.of("successfulAtlasServices", successfullyStoredBackups),
                     SafeArg.of("atlasServicesWithoutBackup", atlasServicesWithInProgressBackups));
         }
 
         return successfulBackups.keySet();
     }
 
-    private void storeCompletedBackup(AtlasService atlasService, CompletedBackup completedBackup) {
-        coordinationServiceRecorder.storeFastForwardState(atlasService, completedBackup);
-        backupPersister.storeCompletedBackup(atlasService, completedBackup);
+    private Set<AtlasService> storeCompletedBackups(Map<AtlasService, CompletedBackup> successfulBackups) {
+        ExecutorService executorService = PTExecutors.newFixedThreadPool(completeBackupNumThreads);
```

Comment:
Is this logic running per datastore? 
Also, can we extract a final variable for executorService and create that only once?

Reply:
No, this logic will run once per timelock service on a deployment.

I'd rather have the executor service be tightly scoped, so that we can shut it down (and thus reclaim the resources) when we've finished with it.